src/MongoDB.Driver.Core/ReadPreference.cs

```diff
@@ -240,18 +251,33 @@ public override string ToString()
                 sb.Append(", MaxStaleness : ");
                 sb.Append(TimeSpanParser.ToString(_maxStaleness.Value));
             }
+            if (_hedge != null)
+            {
+                sb.Append(", Hedge : ");
+                sb.Append(_hedge.ToBsonDocument().ToJson());
+            }
             sb.Append(" }");
             return sb.ToString();
         }
 
+        /// <summary>
+        /// Returns a new instance of ReadPreference with some values changed.
+        /// </summary>
+        /// <param name="hedge">The hedge.</param>
+        /// <returns>A new instance of ReadPreference.</returns>
+        public ReadPreference With(ReadPreferenceHedge hedge)
```

Comment:
Note: this convention of using just `With` as the method name and using the type of the parameter to distinguish what is being changed will break if there are ever two parameters of the same type, which seems quite possible with primitive types like `int` and `string`.

Nevertheless I've named the method just `With` to match the other methods in this class.